src/java-opentimelineio/src/main/java/io/opentimeline/opentimelineio/Transition.java

```diff
@@ -0,0 +1,123 @@
+package io.opentimeline.opentimelineio;
+
+import io.opentimeline.opentime.RationalTime;
+import io.opentimeline.opentime.TimeRange;
+
+public class Transition extends Composable {
+
+    public static class Type {
+        public static String SMPTE_Dissolve = "SMPTE_Dissolve";
+        public static String Custom = "Custom_Transition";
+    }
+
+    protected Transition() {
```

Comment:
any reason this is protected?

Reply:
Originally I didn't want to provide a constructor with no args. So having an empty public constructor was of no use.

If there's a class deriving from Transition it will need to call the constructor of the super class. The constructor in Transition with args would call the `native initialize` method, right? If the derived class had a few more arguments, that class would call it's own `native initialize method` which in turn would allocate memory again.

So I thought it would help to have an empty constructor for super classes, but exposing them publicly would be useless. So I made it protected.

Am I wrong anywhere?

Reply:
Don't think there is anything wrong, just trying to figure out why.
I would typically expect a class extending Transition not to create an empty Transition object using a special constructor but rather call the one with all of the required arguments. One possible concern here is that implementer extending transition not setting the necessary params. If that is not an issue here, we can ignore. My question is though - if we did not have an empty constructor at all, would that be costly?

Reply:
I'll recheck tomorrow if there is any case where all the required arguments are not set. I'm not on my system at the moment. 
 For some objects we would have two or three such layers.
 For a track it could call the constructor of composition, item, composable, serializableobjectwithmetadata and serializableobject.